src/coverlet.core/Helpers/InstrumentationHelper.cs

```diff
@@ -10,21 +10,22 @@
 using System.Text.RegularExpressions;
 using Coverlet.Core.Abstracts;
 using Microsoft.Extensions.FileSystemGlobbing;
-using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.FileSystemGlobbing.Abstractions;
 
 namespace Coverlet.Core.Helpers
 {
-    internal static class InstrumentationHelper
+    internal class InstrumentationHelper : IInstrumentationHelper
     {
-        private static readonly ConcurrentDictionary<string, string> _backupList = new ConcurrentDictionary<string, string>();
+        private readonly ConcurrentDictionary<string, string> _backupList = new ConcurrentDictionary<string, string>();
+        private IRetryHelper _retryHelper;
 
-        static InstrumentationHelper()
+        public InstrumentationHelper(IProcessExitHandler processExitHandler, IRetryHelper retryHelper)
         {
-            DependencyInjection.Current.GetService<IProcessExitHandler>().Add((s, e) => RestoreOriginalModules());
+            processExitHandler.Add((s, e) => RestoreOriginalModules());
```

Comment:
@ViktorHofer you did this change, now a single instance of `InstrumentationHelper`  will be returned by ["static" container](https://github.com/tonerdo/coverlet/pull/531/files#diff-7358e9f0a54c6af47045bebb661bb44eR33)(singleton) so the delegate will keep object alive for "restore".
Is it ok for you or do you see some issue with it?